Kitodo/src/test/java/org/kitodo/production/forms/dataeditor/StructurePanelIT.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * (c) Kitodo. Key to digital objects e. V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo project.
+ *
+ * It is licensed under GNU General Public License version 3 or later.
+ *
+ * For the full copyright and license information, please read the
+ * GPL3-License.txt file that was distributed with this source code.
+ */
+
+package org.kitodo.production.forms.dataeditor;
+
+import java.lang.reflect.Method;
+import java.net.URI;
+import java.util.LinkedList;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.kitodo.data.database.beans.Process;
+import org.primefaces.model.DefaultTreeNode;
+
+public class StructurePanelIT {
+
+    @Test
+    public void testAddParentLinksRecursive() throws Exception {
+        DataEditorForm dummyDataEditorForm = new DataEditorForm();
+        final StructurePanel underTest = new StructurePanel(dummyDataEditorForm);
+
+        Process parent = new Process();
+        parent.setProcessBaseUri(URI.create("2"));
+        Process child = new Process();
+        child.setId(42);
+        child.setParent(parent);
+        LinkedList<DefaultTreeNode> result = new LinkedList<>();
+
+        Method addParentLinksRecursive = StructurePanel.class.getDeclaredMethod("addParentLinksRecursive",
```

Comment:
WHY are you invoking your tested method by reflections???

Reply:
Because `addParentLinksRecursive()` is `private` and cannot be accessed by other means. Trying to call it directly would not even compile.

Reply:
So first main question... Why do you try to access private method instead of test public method in which this private is used...?

Reply:
Kathrin wanted me to write a test for precisely this method. In addition, this method is called by a method that again is `private`.